modules/terraform/plan_info.go

```diff
@@ -0,0 +1,75 @@
+package terraform
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type Attributes map[string]interface{}
+
+type Change struct {
+	Actions      []string
+	Before       Attributes
+	After        Attributes
+	AfterUnknown Attributes `json:"after_unknown"`
+}
+
+// TODO: The `Index` field can be a string or an int depending on if `count` or
+// `for_each` was use.  Doesn't parse as an int right now.
+type Resource struct {
+	Module       string
+	Address      string
+	Mode         string
+	Type         string
+	Name         string
+	Index        string
```

Comment:
NIT: It would help to have a `json` tag on all the attributes that will be extracted, even if it is not necessary. It looks like some of these are fields that are managed by terratest (e.g `Module`), although I might be misunderstanding your code.

Reply:
Yeah, this is sorta messy.  All of these attributes would be extracted. I can put `json` tags on them for sure.  I can also nest these, there's really no reason why anyone would need to use them un-nested (I wasn't aware until recently that I could nest structs in Go)